streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/TimeWindowedKStream.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.scala.kstream
+
+import org.apache.kafka.streams.kstream.{TimeWindowedKStream => TimeWindowedKStreamJ, _}
+import org.apache.kafka.streams.state.WindowStore
+import org.apache.kafka.common.utils.Bytes
+import org.apache.kafka.common.serialization.Serde
+import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.FunctionConversions._
+
+/**
+ * Wraps the Java class TimeWindowedKStream and delegates method calls to the underlying Java object.
+ */
+class TimeWindowedKStream[K, V](val inner: TimeWindowedKStreamJ[K, V]) {
+
+  def aggregate[VR](initializer: () => VR,
+    aggregator: (K, V, VR) => VR): KTable[Windowed[K], VR] = {
+
+    inner.aggregate(initializer.asInitializer, aggregator.asAggregator)
+  }
+
+  def aggregate[VR](initializer: () => VR,
+    aggregator: (K, V, VR) => VR,
+    materialized: Materialized[K, VR, WindowStore[Bytes, Array[Byte]]]): KTable[Windowed[K], VR] = {
+
+    inner.aggregate(initializer.asInitializer, aggregator.asAggregator, materialized)
+  }
+
+  def count(): KTable[Windowed[K], Long] = {
+    val c: KTable[Windowed[K], java.lang.Long] = inner.count()
+    c.mapValues[Long](Long2long(_))
+  }
+
+  def count(store: String, keySerde: Option[Serde[K]] = None): KTable[Windowed[K], Long] = {
```

Comment:
We can add the following:
```
def count(store: String,
    materialized: Materialized[K, Long, WindowStore[Bytes, Array[Byte]]]): KTable[Windowed[K], Long] = { //..
```
But this one also may be useful when the user just needs to pass in the `keySerde`. She need not construct any `Materialized` which is abstracted within the implementation of the API.

Suggestions ?

Reply:
I'd vote for keeping java / scala API consistent, and we are going to remove deprecated APIs in future releases anyway.

In current API we'd only have one additional overload:

```
def count(materialized: Materialized[K, Long, WindowStore[Bytes, Array[Byte]]]): KTable[Windowed[K], Long] = { //..
```

I think for users who do not want to specify the store name at all, they can rely on

```
static Materialized<K, V, S> with(final Serde<K> keySerde, final Serde<V> valueSerde)
```

to still hide the `materialized` parameter with implicit conversion. 

For users who do want to specify the store name, but want to rely on type conversion, we could call `withKeySerde` and `withValueSerde` internally in the implicit conversion so that user only need to give `Materialized.as(storeName)`

Does that work?

Reply:
+1 .. will remove this overload for `count`.